src/runners/RunnerUtil.ml

```diff
@@ -82,21 +83,19 @@ let import_libs names init_file =
         in
         fatal_error @@ mk_error1 errmsg (get_rep (fst l))
       else
-      if List.mem importedl name ~equal:(=) then (ilibs, importedl) else
       let (ilib, ilib_import_map) = import_lib (fst l) in
-      let (ilibs'', importedl'') = importer ilib.elibs ilib_import_map (name :: stack) (name :: importedl) in
+      let ilibs'' = importer ilib.elibs ilib_import_map (name :: stack) in
+      let libnode = { libn = ilib.libs; deps = ilibs'' } in
       (* Order in which we return the list of imported libraries is important. *)
-      (ilibs @ ilibs'' @ [ilib], importedl'')
-    ) ~init:([], importedl) mapped_names
+      (libacc @ [libnode])
```

Comment:
Same comment as before - can we not use `List.fold_right` to avoid this call to `@`?